tools/fastq_provider/fastq_provider.xml

```diff
@@ -0,0 +1,66 @@
+    <description> Scripts and utilities for providing fastqs to workflows. </description>
+    <macros>
+        <token name="@TOOL_VERSION@">0.4.1</token>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">atlas-fastq-provider</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        fetchFastq.sh -f ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR035/ERR035229/ERR035229.fastq.gz -t ERR035229.fastq.gz
+    ]]></command> 
+
+    <inputs>
+        <param name="source" argument="-f" type="text" label="file or uri" help="file or uri" />
+        <param name="target" argument="-t" type="text"  label="target file" help="target file" />
+        <param name="resource" argument="-s" type="select"  label="resource or directory, default 'auto' " help="It uses this variable to decide how to fetch the file, e.g. there’s a specific method for getting results from hca. " optional='true' >
+            <option value="auto" selected="true">auto</option>
+            <option value="ena">ena</option>
+            <option value="sra">sra</option>
+            <option value="hca">hca</option>
+            <option value="dir">dir</option>
+	</param>
+        <param name="retrieval_method" argument="-m" type="text"  label="retrieval method, default 'wget' " value='wget' help="retrieval method, default 'wget' "  optional='true'/>
+        <param name="mode" argument="-p" type="select" format="select" label="public or private, default public" help="public or private, default public" optional='true'>
+            <option value="public" selected="true">public</option>
+            <option value="private">private</option>
+	</param>
+        <param name="library" argument="-l" type="text"  label="library, by default inferred from file name" help="library, by default inferred from file name" optional='true'/>
+        <param name="config_file" argument="-c" type="data" format="atlas-fastq-provider-config.sh" label="config file to override defaults" help="config file to override defaults" optional='true'/>
+        <param name="validate_only" argument="-v" type="boolean" label="validate only, don't download" help="validate only, don't download" optional='true'/>
+        <param name="download_type" argument="-d" type="select" label="download type, fastq or srr" help="download type, fastq or srr" optional='true'>
+            <option value="fastq" selected="true">fastq</option>
+            <option value="srr">srr</option>
+	</param>
+    </inputs>
+
+    <outputs>
+        <data name="outfile" label="${tool.name} on ${on_string}: fastq file" format="fastq.gz"/>
+    </outputs>
+
+    <tests>
+        <test expect_num_outputs="1">
+            <param name="source" value="ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR035/ERR035229/ERR035229.fastq.gz"/>
+            <param name="target" value="ERR035229.fastq.gz"/>
+            <output name="outfile" file="ERR035229.fastq.gz" ftype="fastq.gz"/>
```

Comment:
This here is expecting to find a file in the test-data folder that is named `ERR035229.fastq.gz`. You need to get rid of the `file` field and add another comparison method. Using file means that you expect galaxy to compare the `outfile` output to that file.

Reply:
@pcm32 thanks